### PR TITLE
🎩[BREAKING] Consistent names. Consistent docs. Working samples.

### PR DESCRIPTION
**Pull request by indexzero**
While preparing for an upcoming talk that will include a deep dive into `@exemplar` I saw several _(mostly)_ non-technical changes that are aimed at:

1. Lowering the bar to comprehend what `@exemplar`.
2. Making it more obvious what each `@exemplar/**` package does.
3. Making it easier for potential new consumers contributors to understand the separation of concerns.

With all that in mind this PR does the following:

- [x] Migrate from `mono-repos` to `lerna`. Using more common tooling will _(in theory)_ make it easier to on-board new contributors.
- [x] Rename `@exemplar/storybook` to `@exemplar/storybook-react`.
- [x] Rename `@exemplar/native-storybook` to `@exemplar/storybook-react-native`.
- [x] Merge `./examples` and `./packages/react/examples` into `./packages/exemplar-samples`. This gives us a true end-to-end test (with local `npm link` support) for testing new changes to `@exemplar` itself. 
   - Feel strongly this is important since the `./examples` using the `design-system.js` did not work due to a typo in how the examples were exported as anonymous functional components (vs. named exports).
- [x] Deep scrub all the documentation for consistency and word choice.

### CODE REVIEW

File: packages/exemplar-storybook-react/package.json [hunk 1]
```diff
@@ -39,24 +39,24 @@
...
+    "@storybook/addons": "^5.1.9",
     "@storybook/react": "^5.1.0",
-    "assume": "^2.1.0",
-    "babel-eslint": "^10.0.1",
-    "babel-loader": "^8.0.5",
+    "assume": "^2.2.0",
+    "babel-eslint": "^10.0.2",
+    "babel-loader": "^8.0.6",
     "eslint": "^5.14.1",
     "eslint-config-godaddy-react": "^4.1.0",
```
Comment (msluther) [inline packages/exemplar-storybook-react/package.json]: we should get greenkeeper on here at some point

Comment (indexzero) [reply packages/exemplar-storybook-react/package.json]: I'd prefer renovate, but agreed.

File: packages/exemplar-storybook-react-native/.npmignore [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#
+haul-debug.log
+
+# Ignore generated `storybook/stories/index.js`
+storybook/stories/index.js
+metro.config.js
+
+# The files are added using the `react-native eject` command for development
+ios
+android
```
[the remark this answers is not in the document]
Comment (Swaagie) [reply packages/exemplar-storybook-react-native/.npmignore]: `postinstall`  hook will readd these folders to the published module

File: packages/exemplar-storybook-react-native/package.json [hunk 3]
```diff
@@ -8,8 +8,26 @@
     "test": "nyc mocha test.js"
   },
   "bin": {
-    "exemplar-native": "bin/start"
+    "start-exemplar-native": "bin/start"
```
[the remark this answers is not in the document]
Comment (Swaagie) [reply packages/exemplar-storybook-react-native/package.json]: Im fine with naming it `examplar-react-native` not so sure about prepending `start`, but not feeling about it

File: packages/exemplar-storybook-react-native/.npmignore [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### ▲ .gitignore contents above
+### ▼ additional ignore files for publishing
+
+# Coverage directory used by tools like istanbul
+coverage
+.nyc_output
```
[the remark this answers is not in the document]
Comment (indexzero) [reply packages/exemplar-storybook-react-native/.npmignore]: They are in the root `.gitignore` for the entire repository.

File: packages/exemplar-storybook-react-native/.npmignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (indexzero) [reply packages/exemplar-storybook-react-native/.npmignore]: Thanks @Swaagie – I was able to consume the package published with this `.npmignore` so I was pretty sure it was correct.

File: packages/exemplar-sample/examples/text.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import React from 'react';
 import { Text } from 'super-secret-components';
 
-export default () => (
+const GreenText = () => (
```
[the remark this answers is not in the document]
Comment (indexzero) [reply packages/exemplar-sample/examples/text.js]: We could use `function`, but FYI the name of the assignment is used for arrow functions. e.g.

``` js
const OtherFunction = () => {
  console.log('OtherFunction');
  throw new Error('Trace me.');
}

const NamedFunction = () => {
  console.log('NamedFunction');
  OtherFunction();
}

NamedFunction();
```

outputs:

```
~/wtfstack/index.js:4
  throw new Error('Trace me.');
  ^

Error: Trace me.
    at OtherFunction (~/wtfstack/index.js:4:9)
    at NamedFunction (~/wtfstack/index.js:9:3)
    at Object.<anonymous> (~/wtfstack/index.js:12:1)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
    at startup (internal/bootstrap/node.js:283:19)
```

File: packages/exemplar-sample/package.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@exemplar/sample",
+  "version": "0.0.0",
+  "private": "true",
+  "description": "End to end sample usage for @exemplar/storybook-react{,-native}",
+  "main": "src/design-system.js",
+  "scripts": {
+    "storybook": "start-storybook -c ./node_modules/@exemplar/storybook-react",
```
[the remark this answers is not in the document]
Comment (indexzero) [reply packages/exemplar-sample/package.json]: That's a great question! Here's my philosophical answer: 

- We should try to be a light abstraction around `storybook` when we can unless we get requests from users that it's confusing to do so.
- Adding explicit binaries makes it more difficult to pass through all of the other options that the underlying `*storybook*` binary may expose. 
   - e.g. when `start-storybook` adds `--some-new-option` we need to make a PR to plumb that through in the wrapper.

My understanding was that for React Native we absolutely needed our own binary for pathing reasons, which is a reasonable exception to the philosophical position above. 

Thoughts?

File: packages/exemplar-sample/package.json [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    "storybook:ios": "exemplar-native --platform ios",
+    "storybook:android": "exemplar-native --platform android"
```
[the remark this answers is not in the document]
Comment (indexzero) [reply packages/exemplar-sample/package.json]: Fixed in a81f649

File: packages/exemplar-storybook-react-native/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (indexzero) [reply packages/exemplar-storybook-react-native/package.json]: Added both as potential aliases. I am reminded of how happy I am when `npm isntall` also works 😸

File: packages/exemplar-sample/README.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+`@exemplar` will use this file structure to automatically generate a
+`storybook` for you.
+
+## Usage
+
+To see how `@exemplar` works just install dependencies and run the appropriate
+`npm` script:
+
+```
```
Comment (msluther) [inline packages/exemplar-sample/README.md]:
````suggestion
``` bash
````

File: packages/exemplar-sample/README.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
+# `@exemplar/sample`
+
+The `./examples` directory serves as an exemplar for how you can use 
+`@exemplar` for your component, component library, or design system. We strive
+keep it up-to-spec with the latest developments in `@exemplar`, and use every package available in the `@exemplar` ecosystem. 
+
+The base structure of this directory as follows:
+
+``` bash
```
Comment (msluther) [inline packages/exemplar-sample/README.md]: this one isn't really bash scripts
````suggestion
```
````

File: packages/exemplar-sample/package.json [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  },
+  "homepage": "https://github.com/godaddy/exemplar/tree/master/packages/exemplar-sample",
+  "dependencies": {
+    "@exemplar/storybook-react": "*",
+    "@exemplar/storybook-react-native": "*",
+    "@storybook/addons": "^5.1.9",
+    "@storybook/react": "^5.1.0",
+    "prop-types": "^15.7.2",
+    "react": "^16.8.3",
+    "react-dom": "16.8.3",
```
Comment (msluther) [inline packages/exemplar-sample/package.json]:
```suggestion
    "react-dom": "^16.8.3",
```

File: packages/exemplar-storybook-react/CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+# CHANGELOG
+
+### 1.0.0
+
+- Rename from `@exemplar/storybook` for clarity around purpose & scope.
```
Comment (msluther) [inline packages/exemplar-storybook-react/CHANGELOG.md]:
```suggestion
- Rename from `@exemplar/storybook-react` for clarity around purpose & scope.
```

**Comment by indexzero** [comment]
CI failures are due to package name changes. Will ensure that CI passed before publish.

**Pull request merged**